packages/rum-core/src/browser/performanceCollection.ts

```diff
@@ -154,6 +161,17 @@ export function startPerformanceCollection(lifeCycle: LifeCycle, configuration:
   }
 }
 
+function waitForRequestIdleOrUnload(lifeCycle: LifeCycle, callback: () => void) {
+  let hasBeenCalled = false
+  const callOnce = () => {
+    if (hasBeenCalled) return
+    hasBeenCalled = true
+    callback()
+  }
+  requestIdleCallback(callOnce)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏maybe we should add some logic to make sure they are sent before a new view is created?

Reply:
I suppose your fear is that the resources are being associated to the wrong view? 
The only case I can think of, would be to have a location change before the setTimeout callback is called.
I guess that could happen. 
Considering our last discussion, to me it feels a bit of the same concern that the BEFORE_UNLOAD, meaning: 
it's very unlikely to happen but can technically happen.

Here is the list of our options:
- ignore these cases
- consider all cases:  view update, before_unload, visibility change
- only consider view update
- drop the optimisation

My personal opinion would be in favour of either ignoring these cases or drop the optimisation
Wdyt?